librubyfmt/src/render_targets.rs

```diff
@@ -144,7 +144,9 @@ impl AbstractTokenTarget for BreakableEntry {
     }
 
     fn is_multiline(&self) -> bool {
-        self.line_numbers.len() > 1 || self.any_collapsing_newline_has_heredoc_content()
+        self.line_numbers.len() > 1
+            || self.any_collapsing_newline_has_heredoc_content()
+            || self.contains_hard_newline()
```

Comment:
I also think this was a latent bug we've had around forever -- some expressions are essentially _always_ multiline (e.g. classes), so breakables that have these in them should always render multiline.